app/models/account.rb

```diff
@@ -93,7 +93,7 @@ def with_valid_account_type(valid_account_type)
     if accttype == valid_account_type
       yield
     else
-      raise _("Cannot call method '%{caller}' on an Account of type '%{type}'") % {:caller => caller[0][/`.*'/][1..-2],
+      raise _("Cannot call method '%{caller}' on an Account of type '%{type}'") % {:caller => caller(1..1).first[/`.*'/][1..-2],
                                                                                    :type   => accttype}
```

Comment:
this also looks strange to me

Reply:
correct - this prevents it from returning the entire payload